core/src/main/java/org/elasticsearch/cluster/block/ClusterBlocks.java

```diff
@@ -300,6 +300,15 @@ public Builder addBlocks(IndexMetaData indexMetaData) {
             return this;
         }
 
+        public Builder updateBlocks(IndexMetaData indexMetaData) {
+            removeIndexBlock(indexMetaData.index(), MetaDataIndexStateService.INDEX_CLOSED_BLOCK);
+            removeIndexBlock(indexMetaData.index(), IndexMetaData.INDEX_READ_ONLY_BLOCK);
+            removeIndexBlock(indexMetaData.index(), IndexMetaData.INDEX_READ_BLOCK);
+            removeIndexBlock(indexMetaData.index(), IndexMetaData.INDEX_WRITE_BLOCK);
+            removeIndexBlock(indexMetaData.index(), IndexMetaData.INDEX_METADATA_BLOCK);
+            return addBlocks(indexMetaData);
+        }
+
```

Comment:
I agree that the method name is a bit ambiguous. With respect to "specific blocks", these are exactly the same blocks as the ones added by the `addBlocks` method. As updateBlocks depends directly on implementation details of addBlocks, I put it in the same location.

Both methods could get more specific names, for example `addIndexMedataBlocks` and `updateIndexMetadataBlocks`.
@tlrx @imotov what do you think?